### PR TITLE
[WIP] Reform ci

### PR DESCRIPTION
**Pull request by sljlp**
### PR types

Others
### PR changes

Others
### Describe

reform distributed ci
1. Moving some collective and fleet cases into subdirectories "collective" and "fleet" respectively.
2. Adding two testlists.csv files as configure files for generating CMakeLists.txt files.
3. Using ${PADDLE_ROOT}/tools/gen_ut_cmakelists.py to generate CMakeLsist.txt files in subdirectory collective and fleet.

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/fleet/testslist.csv [hunk 1]
```diff
@@ -0,0 +1,92 @@
+name,os,arch,timeout,run_type,launcher,dist_ut_port,ENVS,conditions
```
Comment (gongweibao) [inline python/paddle/fluid/tests/unittests/fleet/testslist.csv:1]: First line is header.

File: python/paddle/fluid/tests/unittests/collective/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,230 @@
+set(LOCAL_ALL_ARCH ON)
```
Comment (gongweibao) [inline python/paddle/fluid/tests/unittests/collective/CMakeLists.txt:1]: Add lines
`# licenses`
`# WARNING comment`

File: python/paddle/fluid/tests/unittests/collective/testslist.csv [hunk 3]
```diff
@@ -0,0 +1,36 @@
+name,os,arch,timeout,run_type,launcher,dist_ut_port,ENVS,conditions
```
Comment (gongweibao) [inline python/paddle/fluid/tests/unittests/collective/testslist.csv:1]: Add line header comment in docs.

**Review by gongweibao — CHANGES_REQUESTED** [review]
Some comments.

File: python/paddle/fluid/tests/unittests/collective/CMakeLists.txt (hunk 2 above)
Comment (sljlp) [reply python/paddle/fluid/tests/unittests/collective/CMakeLists.txt:1]: 1. no need of licenses

**Pull request closed without merging**